modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php

```diff
@@ -19,6 +19,10 @@ public function __construct( $attachment_image, $needs_attachment_link, $graysca
 		}
 
 		$this->orig_file = wp_get_attachment_url( $this->image->ID );
+		// If Photon is active, use it for original
+		if ( class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' ) ) {
```

Comment:
Why are we checking for the class here? We're using a function that isn't part of the class, so we can check for the function instead? The check shouldn't be required since the photon functions are always loaded, but better to play it defensively.

Reply:
I've submitted a change that checks if the Photon module is active instead of the class.
Sorry about that, intent was to check whether Photon setting was active.
Testing for `function_exists( 'jetpack_photon_url' )` returns true even when Photon is not active. Maybe because [tiled galleries runs it](https://github.com/Automattic/jetpack/blob/master/modules/tiled-gallery/tiled-gallery.php#L146) on `gallery_html`? Which makes it muddy if check is necessary. Though if someone has tiled galleries on and photon off, respecting the setting for the link at least seems like the right thing?

Reply:
Yup, `jetpack_photon_url` is always available since there are some cases where we need dynamic image sizes (tiled galleries, the thumbnails for Related Posts, etc), so checking for the module makes sense.

Since displaying the full image doesn't require Photon to operate, checking for site administrator intent with the module check makes sense to me.